### PR TITLE
Updated the description of proposal states

### PR DESCRIPTION
**Pull request by OndroMih**
Proposing update to the description of the proposal/specification states, according to the discussion on the mailing list: https://groups.google.com/forum/#!topic/microprofile/gtUUnntbPtk

List of changes:
 - clarified (and simplified) the *Awaiting review* and *Active review* states 
 - slightly improved the *Returned for revision* state
 - updated and corrected the *Accepted* state
 - removed the *Accepted with revisions* as it doesn't make sense to have 2 different "Accepted" states
 - clarified the *Rejected* state
 - split the *Implemented* state into *Implemented* and *Included in MicroProfile* states

### CODE REVIEW

**Review by heiko-braun — APPROVED** [review]
Looks good to me

File: process.md [hunk 1]
```diff
@@ -78,27 +78,23 @@ repository](https://github.com/microprofile/evolution) will be updated to reflec
 ## Proposal states
 A given proposal can be in one of several states:
 
-* **Awaiting review**: The proposal is awaiting review. Once known,
-  the dates for the actual review will be placed in the proposal
-  document and updated in the [list of proposals](proposals.xml). When the
-  review period begins, the review manager will update the state to
+* **Awaiting review**: The proposal is awaiting review. It should be added into the [list of proposals](https://github.com/eclipse/microprofile/tree/master/proposals). When the
+  review period is announced (no the mailing list), the state transitions to
```
Comment (pilhuhn) [inline process.md]: *on* the ML ?

**Comment by johnament** [comment]
Several of the sections mention "core team" - we need a term to replace that.

**Comment by mikecroft** [comment]
@johnament IIRC the "core team" was mostly just to refer to the most active people. I think most of those are represented in the Eclipse project committers list?
https://projects.eclipse.org/projects/technology.microprofile/who

That's probably the easiest change to make, since it's pretty obvious who it refers to, anyone can look it up, and it can easily be updated independently of this process doc.

**Comment by mikecroft** [comment]
I created issue #29 for the core team change, because I think that's separate to this PR and shouldn't stop this particular change being merged.

**Review by mikecroft — APPROVED** [review]

**Pull request merged**